community/command-line/src/main/java/org/neo4j/commandline/admin/AdminCommand.java

```diff
@@ -81,9 +81,9 @@ public String name()
         public abstract String summary();
 
         /**
-         * @return AdminCommandSegment the command using the provider is grouped under
+         * @return AdminCommandSection the command using the provider is grouped under
          */
-        public abstract AdminCommandSegment segment();
+        public abstract AdminCommandSection commandSection();
```

Comment:
I think you're making this more complex than it has to be. Why not just have this method return a straight string? And have a default implementation returning `General`?